app/models/miq_queue.rb

```diff
@@ -623,6 +623,15 @@ def self.display_name(number = 1)
 
   private
 
+  # NOTE: this will lookup an unknown zone every time.
+  def validate_zone_name
+    unless zone.nil? || self.class.valid_zone_names[zone] ||= Zone.in_my_region.exists?(:name => zone)
+      errors.add(:zone, N_("Unknown zone"))
+    end
+  end
+
+  cache_with_timeout(:valid_zone_names) { Hash.new }
```

Comment:
Perhaps we can cache with a smaller timeout (default is 5 minutes)? Or, perhaps we cache thread locally instead of globally?

Reply:
My concern is that if a zone is deleted, the cache will still be present and allow queue items for an "old" zone.  Technically that's bound to happen anyway regardless of what timeout we pick, but lowering the timeout number minimizes that problem.

Reply:
Oh, zone _is_ the name - so strange we don't use zone_id :(

Reply:
yes. so strange